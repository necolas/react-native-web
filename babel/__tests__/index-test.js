const plugin = require('..');
const pluginTester = require('babel-plugin-tester');

pluginTester({
  plugin,
  snapshot: true,
  tests: [
    // import react-native
    "import { View } from 'react-native';",
    "import { Switch, Text, View as MyView, ViewPropTypes } from 'react-native';",
    "import { createElement, Switch, StyleSheet } from 'react-native';",
    "import { InvalidThing, TouchableOpacity } from 'react-native';",
    "import * as RNW from 'react-native';",

    // import react-native-web
    // "import { View } from 'react-native-web';",
    // "import { Switch, Text, View as MyView } from 'react-native-web';",
    // "import { createElement, Switch, StyleSheet } from 'react-native-web';",
    // "import { InvalidThing, TouchableOpacity } from 'react-native-web';",
    // "import * as RNW from 'react-native-web';",

    // require react-native
    "const { View } = require('react-native');",
    "let { Switch, Text, View: MyView } = require('react-native');",
    "var { createElement, Switch, StyleSheet } = require('react-native');",
    "const { InvalidThing, TouchableOpacity } = require('react-native');",

    // require react-native-web
    // "const { View } = require('react-native-web');",
    // "let { Switch, Text, View: MyView } = require('react-native-web');",
    // "var { createElement, Switch, StyleSheet } = require('react-native-web');",
    // "const { InvalidThing, TouchableOpacity } = require('react-native-web');",

    // export react-native
    "export { View } from 'react-native';",
    "export { Switch, Text, View as MyView, ViewPropTypes } from 'react-native';",
    "export { createElement, Switch, StyleSheet } from 'react-native';",
    "export { InvalidThing, TouchableOpacity } from 'react-native';",
    "export { default as RNW } from 'react-native';",
    {
      code: "const RNW = require('react-native');",
      output: "const RNW = require('react-native');",
      snapshot: false
    }
  ]
});
