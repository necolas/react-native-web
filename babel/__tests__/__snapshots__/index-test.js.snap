// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1. Rewrite react-native paths for react-native-web 1`] = `
"
import { View } from 'react-native';

      ↓ ↓ ↓ ↓ ↓ ↓

import View from 'react-native-web/dist/components/View';
"
`;

exports[`2. Rewrite react-native paths for react-native-web 1`] = `
"
import { Switch, Text, View as MyView, ViewPropTypes } from 'react-native';

      ↓ ↓ ↓ ↓ ↓ ↓

import Switch from 'react-native-web/dist/components/Switch';
import Text from 'react-native-web/dist/components/Text';
import MyView from 'react-native-web/dist/components/View';
import ViewPropTypes from 'react-native-web/dist/components/View/ViewPropTypes';
"
`;

exports[`3. Rewrite react-native paths for react-native-web 1`] = `
"
import { createElement, Switch, StyleSheet } from 'react-native';

      ↓ ↓ ↓ ↓ ↓ ↓

import createElement from 'react-native-web/dist/modules/createElement';
import Switch from 'react-native-web/dist/components/Switch';
import StyleSheet from 'react-native-web/dist/apis/StyleSheet';
"
`;

exports[`4. Rewrite react-native paths for react-native-web 1`] = `
"
import { InvalidThing, TouchableOpacity } from 'react-native';

      ↓ ↓ ↓ ↓ ↓ ↓

import { InvalidThing } from 'react-native-web';
import TouchableOpacity from 'react-native-web/dist/components/Touchable/TouchableOpacity';
"
`;

exports[`5. Rewrite react-native paths for react-native-web 1`] = `
"
import * as RNW from 'react-native';

      ↓ ↓ ↓ ↓ ↓ ↓

import * as RNW from 'react-native-web';
"
`;

exports[`6. Rewrite react-native paths for react-native-web 1`] = `
"
const { View } = require('react-native');

      ↓ ↓ ↓ ↓ ↓ ↓

const View = require('react-native-web/dist/components/View');
"
`;

exports[`7. Rewrite react-native paths for react-native-web 1`] = `
"
let { Switch, Text, View: MyView } = require('react-native');

      ↓ ↓ ↓ ↓ ↓ ↓

let Switch = require('react-native-web/dist/components/Switch');

let Text = require('react-native-web/dist/components/Text');

let MyView = require('react-native-web/dist/components/View');
"
`;

exports[`8. Rewrite react-native paths for react-native-web 1`] = `
"
var { createElement, Switch, StyleSheet } = require('react-native');

      ↓ ↓ ↓ ↓ ↓ ↓

var createElement = require('react-native-web/dist/modules/createElement');

var Switch = require('react-native-web/dist/components/Switch');

var StyleSheet = require('react-native-web/dist/apis/StyleSheet');
"
`;

exports[`9. Rewrite react-native paths for react-native-web 1`] = `
"
const { InvalidThing, TouchableOpacity } = require('react-native');

      ↓ ↓ ↓ ↓ ↓ ↓

const TouchableOpacity = require('react-native-web/dist/components/Touchable/TouchableOpacity');
"
`;

exports[`10. Rewrite react-native paths for react-native-web 1`] = `
"
export { View } from 'react-native';

      ↓ ↓ ↓ ↓ ↓ ↓

export { default as View } from 'react-native-web/dist/components/View';
"
`;

exports[`11. Rewrite react-native paths for react-native-web 1`] = `
"
export { Switch, Text, View as MyView, ViewPropTypes } from 'react-native';

      ↓ ↓ ↓ ↓ ↓ ↓

export { default as Switch } from 'react-native-web/dist/components/Switch';
export { default as Text } from 'react-native-web/dist/components/Text';
export { default as MyView } from 'react-native-web/dist/components/View';
export { default as ViewPropTypes } from 'react-native-web/dist/components/View/ViewPropTypes';
"
`;

exports[`12. Rewrite react-native paths for react-native-web 1`] = `
"
export { createElement, Switch, StyleSheet } from 'react-native';

      ↓ ↓ ↓ ↓ ↓ ↓

export { default as createElement } from 'react-native-web/dist/modules/createElement';
export { default as Switch } from 'react-native-web/dist/components/Switch';
export { default as StyleSheet } from 'react-native-web/dist/apis/StyleSheet';
"
`;

exports[`13. Rewrite react-native paths for react-native-web 1`] = `
"
export { InvalidThing, TouchableOpacity } from 'react-native';

      ↓ ↓ ↓ ↓ ↓ ↓

export { InvalidThing } from 'react-native-web';
export { default as TouchableOpacity } from 'react-native-web/dist/components/Touchable/TouchableOpacity';
"
`;

exports[`14. Rewrite react-native paths for react-native-web 1`] = `
"
export { default as RNW } from 'react-native';

      ↓ ↓ ↓ ↓ ↓ ↓

export { default as RNW } from 'react-native-web';
"
`;
